moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -163,7 +163,7 @@ void RobotState::copyFrom(const RobotState& other)
             (robot_model_->getVariableCount() * (1 + ((has_velocity_ || has_acceleration_ || has_effort_) ? 1 : 0) +
                                                  ((has_acceleration_ || has_effort_) ? 1 : 0)) +
              nr_doubles_for_dirty_joint_transforms);
-    memcpy(variable_joint_transforms_, other.variable_joint_transforms_, bytes);
+    memcpy((char*)variable_joint_transforms_, (char*)other.variable_joint_transforms_, bytes);
```

Comment:
Is this really a fix? memcpy() can break alignment, no? I also ran into this and would have thought that we should move to stl types for these or use a [Map](https://eigen.tuxfamily.org/dox/classEigen_1_1Map.html).

Reply:
This was tuned (though several years ago) to optimize CPU cache hit rates. To this end, it does make sense to keep all the robot state in a single memory block. I don't see how Eigen::Map would help here. It's fine to utilize this memory block directly for Eigen::Isometry. The (new) gcc warning I fixed here is related to the fact that we memcpy the content of Eigen::Isometry, thus by-passing the custom-defined copy-constructor of Eigen::Isometry.
As this copy-constructor just performs a simple check, we are safe to perform the memcpy. Casting to (char*), we pretend (for gcc) that this isn't a Eigen::Isometry array :wink:

Reply:
```suggestion
    memcpy((void*)variable_joint_transforms_, (void*)other.variable_joint_transforms_, bytes);
```
Officially `memcpy` receives `void*` arguments, so there should be no need for the `char` hack.